### ISSUE TITLE
Accessibility API/WCAG compliance

### ISSUE THREAD

**Issue by gselderslaghs** [materializecss/materialize#543]
### Before submitting...

- [X] I have searched for [duplicate or closed issues](https://github.com/materializecss/materialize/issues?utf8=%E2%9C%93&q=is%3Aissue).
- [X] I have read the [CONTRIBUTING document](https://github.com/materializecss/materialize/blob/master/CONTRIBUTING.md) and my issue is following the guidelines.
- [X] I have read the template completely before filling it in.

### Context

This issue is the epic referencing multiple accessibility issues within the Materialize framework

### Current Behavior

I've found many items already are compliant with accessibility standards and would like to thank the community for working on those items  🚀

As of today, multiple components are still missing accessibility implementations and as result his causes a web accessibility reduction regarding accessibility API and WCAG compliance

### Expected behavior

Materialize should be accessibile to
- People who rely on keyboard control
- People with dexterity impairments using voice control or a keyboard-like switch device
- People with low vision who use screen magnification with keyboard control
- People using a screen reader, including blind, low vision, and neurodivergent people
- Neurodivergent people who use the keyboard
- Older adults who use the keyboard

### Possible Solutions or Causes

In order to align with accessibility API and WCAG compliance the following issues should be addressed and validated using accessibility checker tools

- [ ] #81 
- [ ] #246 
- [ ] #510 
- [ ] #522 
- [ ] #538 
- [ ] #539 
- [ ] #540 
- [ ] #541 

### Your Environment

* Version used: 2.1.1
